src/Http/Http.Abstractions/src/ConnectionInfo.cs

```diff
@@ -8,23 +8,45 @@
 
 namespace Microsoft.AspNetCore.Http
 {
+    /// <summary>
+    /// Represents the underlying connection for a request.
+    /// </summary>
     public abstract class ConnectionInfo
     {
         /// <summary>
         /// Gets or sets a unique identifier to represent this connection.
         /// </summary>
         public abstract string Id { get; set; }
 
+        /// <summary>
+        /// Gets or sets the IP address of the remote target. Can be null.
+        /// </summary>
         public abstract IPAddress? RemoteIpAddress { get; set; }
 
+        /// <summary>
+        /// Gets or sets the port of the remote target.
+        /// </summary>
         public abstract int RemotePort { get; set; }
 
+        /// <summary>
+        /// Gets or sets the IP address of the local host.
+        /// </summary>
         public abstract IPAddress? LocalIpAddress { get; set; }
 
+        /// <summary>
+        /// Gets or sets the port of the local host.
+        /// </summary>
         public abstract int LocalPort { get; set; }
 
+        /// <summary>
+        /// Gets or sets the client certificate.
+        /// </summary>
         public abstract X509Certificate2? ClientCertificate { get; set; }
 
+        /// <summary>
+        /// Retrieves the client certificate.
+        /// </summary>
+        /// <returns>Asynchronously returns a <see cref="X509Certificate2" />. Can be null.</returns>
```

Comment:
```suggestion
        /// <returns>Asynchronously returns an <see cref="X509Certificate2" />. Can be null.</returns>
```

Reply:
I think "a" is right here since X isn't a vowel unless you're using some other heuristic here.

Reply:
https://blog.apastyle.org/apastyle/2012/04/using-a-or-an-with-acronyms-and-abbreviations.html

X is pronounce as "ex", so we want "returns an ex-509Certicate2"

Yay English 😄